src/Symfony/Component/Config/Loader/ParamConfigurator.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Config\Loader;
```

Comment:
`\Definition` IMHO

Reply:
Actually, the ParamConfigurator itself still looks tied to the DI component, as it implements the DI convention of `%param_name%`.

If the reason to move it to symfony/config is only because generated classes need to reference it in their docblock to allow such type to be passed, another solution would be to define an interface in symfony/config (used in the docblock) that would then be implemented by ParamConfigurator and EnvConfigurator in DI (btw, this would also allow avoiding to extend ParamConfigurator in EnvConfigurator while it works totally differently and does not initialize the parent properties)

Reply:
i thought about it ... we can reason `key: '%value%'` is already ia Config component concern, simply for being a user value we process. Hence i'd be OK with this layering ... but yes it generalizes a certain format - the DI one.

Next, im not sure being able to disallow EnvConfigurators for eg. nodes that dont support placeholders is a worthy goal really - it'll be validated compile time anyway.

Maybe `Stringable|...` is sufficient, i like the abstraction it brings :)

Reply:
> another solution would be to define an interface in symfony/config (used in the docblock) that would then be implemented by ParamConfigurator and EnvConfigurator in DI

Yeah, But that require us to have an interface on a value object. We rarely do that.

Im not sure about using `Stringable`. If a node expects a boolean it looks weird to type hint `bool|Stringable`. It is much better to use a more specific class/interface.

Reply:
at least Stringable gives us compatibility out-of-the-box (no extra symbols needed) with any possible implementation out there.

bool|Value (Value extends Stringable) maybe for extra meaning :)

Reply:
personally id KISS and consider anything Stringable a potential value.

Reply:
> bool|Value (Value extends Stringable) maybe for extra meaning :)

That is what we currently are doing.. But with the object since we dont want to have interfaces for value objects.

Reply:
I suggested this design, and I think it's fine :D
The maybe-coupling to DI is easily overridden when one will need to thanks to inheritance.
I wouldn't go with `Stringable` as it doesn't convey the intent correctly IMHO.
And adding an interface looks overkill to me, and something I prefer avoiding for a VO.

Reply:
let' be pramatic then. What about the namespace?

Reply:
In DI and Routing, the namespace is `Loader\Configurator`; We might add `Configurator` here also. I've strong opinion on the topic.

Reply:
Right, i've overlooked Configurator being a sub of Loader, thinking of siblings instead.

Let's do it like this :+1: sorry for the noise :)

Reply:
@ro0NL the `Config\Definition` component does **not** know about `key: %value%`. Parameters are resolved by the DI component before calling the DI extension and so the Configuration class never sees them.

Reply:
yes, i phrased it wrong. ParamConfigurator is an ordinary value from the component POV.